roles/openshift_health_checker/openshift_checks/mixins.py

```diff
@@ -2,17 +2,14 @@
 from openshift_checks import get_var
 
 
-class NotContainerized(object):
+class NotContainerizedMixIn(object):
```

Comment:
That doesn't work for what we want here, because we don't want to create a class for every combination of the ways how one can determine whether a check is active or not.

Right now we only have `NotContainerized` implemented, but the stage is set to have other things like `EnterpriseOnly`, `ContainerizedOnly`, or such.

If not using Mixins, the next level of sophistication I'd go into is having something like a list of `Condition`s, and evaluating them. Something I can think of, thought is in a different level of sophistication and implementation is https://docs.djangoproject.com/en/1.10/ref/models/conditional-expressions/, but more generally a way to combine functions that given some input tell whether a check applies or not...

Dreaming...

```python

class PackageAvailability(OpenShiftCheck):

    name = "package_availability"
    tags = ["preflight"]

    constraints = [NotContainerized, Origin, ..., ...]

    # ...

# in the check runner:

check_class = PackageAvailability
if all(constraint(...) for constraint in check_class.constraints):
     result = check_class(...).run(...)
```